src/containers/org/OrgRoleContainer.js

```diff
@@ -69,47 +78,81 @@ const OrgRoleContainer = ({
-            <Crumbs>
-              <CrumbLink to={organizationRoute}>{organization.title || 'Organization'}</CrumbLink>
-              <CrumbLink to={rolesRoute}>Roles</CrumbLink>
-              <CrumbItem>{role.title}</CrumbItem>
-            </Crumbs>
-            <SpaceBetweenGrid>
-              <Typography gutterBottom variant="h1">{role.title}</Typography>
-              <RoleActionButton organization={organization} role={role} />
-            </SpaceBetweenGrid>
+      <>
+        <AppContentGrid isVisiblePanelColumn={!!selection}>
+          <ContentColumn>
+            <AppContentWrapper>
+              <Crumbs>
+                <CrumbLink to={organizationRoute}>{organization.title || 'Organization'}</CrumbLink>
+                <CrumbLink to={rolesRoute}>Roles</CrumbLink>
+                <CrumbItem>{role.title}</CrumbItem>
+              </Crumbs>
+              <StackGrid gap={24}>
+                <StackGrid>
+                  <SpaceBetweenGrid>
+                    <Typography variant="h1">{role.title}</Typography>
+                    <RoleActionButton organization={organization} role={role} />
+                  </SpaceBetweenGrid>
+                  {
+                    isNonEmptyString(role.description) && (
+                      <Typography variant="body1">{role.description}</Typography>
+                    )
+                  }
+                </StackGrid>
+                <StackGrid>
+                  <Typography variant="h2">Data Sets</Typography>
+                  <Typography>Click on a dataset to manage permissions.</Typography>
+                </StackGrid>
+                <PermissionsActionsGrid
+                    assignPermissionsText="Add dataset"
+                    onChangeFilterByPermissionTypes={setFilterByPermissionTypes}
+                    onChangeFilterByQuery={setFilterByQuery}
+                    onClickAssignPermissions={() => setIsVisibleAddDataSetModal(true)} />
+                <DataSetPermissionsContainer
+                    filterByPermissionTypes={filterByPermissionTypes}
+                    filterByQuery={filterByQuery}
+                    organizationId={organizationId}
+                    onSelect={setSelection}
+                    principal={role.principal}
+                    selection={selection} />
+              </StackGrid>
+            </AppContentWrapper>
+          </ContentColumn>
+          {
+            selection && (
+              <PanelColumn>
+                <PermissionsPanel
+                    dataSetId={selection.dataSetId}
+                    key={`${selection.dataSetId}-${selection.permissionType}`}
+                    onClose={handleOnClosePermissionsPanel}
+                    principal={role.principal}
+                    permissionType={selection.permissionType} />
+              </PanelColumn>
+            )
+          }
+        </AppContentGrid>
+        <Modal
+            isVisible={isVisibleAddDataSetModal}
+            onClose={() => {}}
```

Comment:
why is this onClose callback empty?

Reply:
For this modal, I'm passing `withHeader={false}` and using my own header component, so `onClose` is a noop because it doesn't do anything. Perhaps this should be a flow fix in LUK.